cmake/Testing.cmake

```diff
@@ -11,7 +11,7 @@ IF(NOT DEFINED ENV{BUILD_ARCHITECTURE} AND EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/te
     target_link_libraries(${UNIT_TEST} PUBLIC 
         ${PROJECT_NAME}
         ${TEST_LINK_LIBRARIES}
-        gtest)
+        GTest::gtest_main)
```

Comment:
Two questions if you have the answer here. What is the difference between Gtest::gtest and GTest::gtest_main. Stack overflow says the difference is that gtest_main comes with a default main function but cmake documentation seems to suggest that there is allows some thread implementation that gets linked to one and not the other.

Reply:
Yeah I understood that part. Was more curious about the other part that mentions linking thread or not but maybe the documentation there is just misleading. Either way this is addressed now.